### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NikitaNikolaitsev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,4 +2,61 @@ class Cargo:
...
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self, name: str, weight: int, coords: list = None
+    ) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords else [0, 0]
```
Comment (DevSpaciX) [inline app/main.py:12]: You can use operator `or` and it will be more cleaner imho

File: app/main.py [hunk 2]
```diff
@@ -2,4 +2,61 @@ class Cargo:
...
+
+class DeliveryDrone(FlyingRobot):
+
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            max_load_weight: int,
+            current_load: int,
+            coords: list = None
```
Comment (DevSpaciX) [inline app/main.py:50]: 1.Also think what type coords can be besides list
2.List of what data type can be ?

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (NikitaNikolaitsev) [reply app/main.py:50]: I have no ideas for implementing

File: app/main.py [hunk 3]
```diff
@@ -2,4 +2,61 @@ class Cargo:
...
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self, name: str, weight: int, coords: list = None  # IDK
+    ) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords or [0, 0]  # you mean this?
```
Comment (DevSpaciX) [inline app/main.py:12]: Yeah , I mean this

File: app/main.py (hunk 2 above)
Comment (DevSpaciX) [reply app/main.py:50]: I mean `coords : list[int] | None = None` : )

File: app/main.py [hunk 4]
```diff
@@ -2,4 +2,61 @@ class Cargo:
...
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        super().__init__(name, weight, coords if coords else [0, 0, 0])
```
Comment (DevSpaciX) [inline app/main.py:33]: Why you did not  use `or` like on line 12 ?

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

**Review by DevSpaciX — APPROVED** [review]
Great !

**Pull request closed without merging**